### PR TITLE
Added cookbook to increment versions across all plugins.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

This is what I had to do to make a 1.2.3 build.
 
### Issues Resolved

Coming from https://github.com/opensearch-project/opensearch-build/issues/1365.
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by peternied — DISMISSED** [review]
Thanks for writing this all up

**Comment by dblock** [comment]
This is now complete given that https://github.com/opensearch-project/opensearch-build/pull/1380 passed all tests. Ready for CR/merge @peternied @saratvemulapalli.

**Comment by dblock** [comment]
@saratvemulapalli @peternied  @peterzhuamazon Do you know/think release notes is also something one person can do across all plugins? If so, can you please help me with a cookbook?

**Comment by davidlago** [comment]
@dblock for release notes, this is what we do in the security plugin (using 1.2.3 as an example to make the step-by-step explanation easier):

1. Get a diff from `1.2` branch's `HEAD` and tag `1.2.2.0`
2. Figure out the list of PRs that were merged in that window
3. Classify them between features, bug fixes and issues (sometimes this can be inferred from the related issue's tags, but this is mostly a manual step for now. If we were to automate this, we would have to be able to fully trust 1) all PRs have an associated issue and 2) all issues have (among others) a tag that helps with this classification. This is not true today.
4. Create a new release notes file (this could be easily "templatized") with a list of changes grouped by category
5. PR out to `main` with the release notes file. This will be cherry-picked into `1.2` before creating the `1.2.3.0` tag

File: META.md [hunk 1]
```diff
@@ -96,3 +108,135 @@ meta exec "git commit -s -m 'Removed integtest.sh.'"
 meta exec "git push username remove-integtest-sh"
 meta exec "gh pr create --title 'Removing default integtest.sh.' --body='Coming from https://github.com/opensearch-project/opensearch-build/issues/497, removing default integtest.sh.'"
 ```
+
+### Increment a Version in Every Plugin
+
+Because one cannot install an older patch version of a plugin on a newer OpenSearch-min (see [OpenSearch#1707](https://github.com/opensearch-project/OpenSearch/issues/1707)), it's common to have to increment versions in all plugins, without making other changes. This was the case in [the 1.2.3 release](https://github.com/opensearch-project/opensearch-build/issues/1365).
```
Comment (bbarani) [inline META.md:114]: The wording should be genetic to any old version of plugins and not specific to patch version

**Review by bbarani — APPROVED** [review]

File: META.md [hunk 2]
```diff
@@ -96,3 +108,135 @@ meta exec "git commit -s -m 'Removed integtest.sh.'"
...
+find . -wholename "*/.github/workflows/*.yml" -print0 | xargs -0 sed -i "s/1.2.2-SNAPSHOT/1.2.3-SNAPSHOT/g"
+find . -wholename "*/.github/workflows/*.yml" -print0 | xargs -0 sed -i "s/1.2.2.0-SNAPSHOT/1.2.3.0-SNAPSHOT/g"
+```
+
+Plugins such as k-nn and security need some exact version replacements.
+
+```
+find . -name build.gradle -print0 | xargs -0 sed -i "s/1.2.2/1.2.3/g"
+find . -name CMakeLists.txt -print0 | xargs -0 sed -i "s/1.2.2.0/1.2.3.0/g"
+find . -name pom.xml -print0 | xargs -0 sed -i "s/1.2.2/1.2.3/g"
```
Comment (davidlago) [inline META.md:150]: Beware of blanked find/replace statements. In security, for example, bumping from 1.2.3 to 1.2.4 would have also matched this https://github.com/opensearch-project/security/blob/main/pom.xml#L81

Comment (dblock) [reply META.md:150]: Fair enough! I think it’s a poor man’s implementation anyway. What we really need is GHA workflows in each repo that increment versions based on created tags.

Comment (dblock) [reply META.md:150]: Here are two attempts at that:

https://github.com/opensearch-project/OpenSearch/pull/1816
https://github.com/opensearch-project/common-utils/pull/106

**Comment by dblock** [comment]
@saratvemulapalli @peternied Can I please get a CR on this?

File: META.md [hunk 3]
```diff
@@ -96,3 +108,135 @@ meta exec "git commit -s -m 'Removed integtest.sh.'"
...
+Create a new manifest that only contains OpenSearch, e.g. [opensearch-build#1369](https://github.com/opensearch-project/opensearch-build/pull/1369). After this manifest is merged, wait for a successful SNAPSHOT build.
+
+#### Increment Version in Plugins
+
+Check out and update the 1.2 branch.
+
+```
+meta git update
+meta git pull origin
+meta git checkout 1.2
```
Comment (saratvemulapalli) [inline META.md:133]: Not a blocker for this PR, just a thought.
I think having an ENV variable for these versions might make it easier when we run this for future versions.

**Review by saratvemulapalli — APPROVED** [review]
Thanks @dblock this is really helpful.
Dropped a minor comment.

File: META.md (hunk 3 above)
Comment (dblock) [reply META.md:133]: This is a good idea, did it anyway.

**Comment by dblock** [comment]
@saratvemulapalli @peternied Re-check please?

**Review by saratvemulapalli — APPROVED** [review]

**Review by peternied — APPROVED** [review]

**Pull request merged**